prov/efa/src/rxr/rxr_init.c

```diff
@@ -429,6 +429,30 @@ int rxr_get_lower_rdm_info(uint32_t version, const char *node,
 	return ret;
 }
 
+static int rxr_shm_space_check(void)
+{
+	struct statvfs stat;
+	char shm_fs[] = "/dev/shm";
+	uint64_t size;
+	int ret = 0;
+
+	ret = statvfs(shm_fs, &stat);
+	if (ret) {
+		FI_WARN(&rxr_prov, FI_LOG_CORE,
+			"Get filesystem %s statistics failed.\n"
+			"shm transfer will be disabled.\n", shm_fs);
```

Comment:
we can ```return err``` here.

Reply:
The actual reason for the stat failing is lost in the callpath. At a minimum, print errno w/strerror here and move the warning about disabling shm to the caller.